packages/commonwealth/client/scripts/views/components/component_kit/cw_table.tsx

```diff
@@ -0,0 +1,222 @@
+import { CWIcon } from './cw_icons/cw_icon';
+import 'components/component_kit/cw_table.scss';
+import { Avatar } from '../Avatar';
+import { ComponentType } from './types';
+
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+
+type ColumnDescriptor = {
+  key: string;
+  header: string;
+  numeric: boolean;
+  sortable: boolean;
+}
+
+type RowData = {
+  avatars?: object;
+}
+
+type TableProps = {
+  columnInfo: ColumnDescriptor[];
+  rowData: any[];
+};
+
+export const CWTable = ({ columnInfo, rowData }: TableProps) => {
+  const [sorting, setSorting] = React.useState<SortingState>([
+    {
+      id: columnInfo[0].key,
+      desc: false
+    }
+  ])
+
+  const [data, _setData] = React.useState(() => rowData)
+
+  const columns = React.useMemo<ColumnDef<unknown, any>[]>(
+    () =>
+      columnInfo.map((col) => {
+        return (
+          {
+            accessorKey: col.key,
+            header: col.header,
+            cell: (info) => {
+              const currentRow = info.row.original as RowData;
+              const avatarUrl = currentRow.avatars?.[col.key];
+
+              if (col.numeric) {
+                return (
+                  <div className='numeric'>
+                    {info.getValue()}
+                  </div>
+                )
+              } else if(avatarUrl){
+                return (
+                  <div className='avatar-cell'>
+                    <Avatar url={avatarUrl} size={20} />
+                    <div className='text'>{info.getValue()}</div>
+                  </div>
+                )
+              } else {
+                return info.getValue()
+              }
+            },
+            footer: props => props.column.id,
+            enableSorting: col.sortable,
+          }
+        )
+      }),
+    []
+  )
+
+  const table = useReactTable({
+    data,
+    columns,
+    state: {
+      sorting,
+    },
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    debugTable: true,
+  })
+
+  const displaySortIcon = (sortDirection: string) => {
+    const sortDirections = {
+      asc: <div className='icon-container'><CWIcon
+              iconName='arrowUpBlue500'
+              iconSize='small'
+              className='arrow-up-blue'
+            /></div>,
+      desc: <div className='icon-container'><CWIcon
+              iconName='arrowDownBlue500'
+              iconSize='small'
+              className='arrow-down-blue'
+              /></div>,
+      false: <div className='icon-container'><CWIcon
+                iconName='arrowUpNeutral400'
+                iconSize='small'
+                className='arrow-down-blue'
+              /></div>,
+    }
+
+    return sortDirections[sortDirection]
+  }
+
+  return (
+    <div className={ComponentType.Table}>
+      <div className="h-2" />
```

Comment:
is this div wrapper necessary? Can't see `h-2` class anywhere defined in scss. Looks like leftover from some tailwind copied code

Reply:
Deleted!